packages/hoprd/crates/hoprd-misc/src/network.rs

```diff
@@ -0,0 +1,32 @@
+use proc_macro_regex::regex;
+
+regex!(is_dns_address_regex "^(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\\.)*[a-z0-9][a-z0-9-]{0,61}[a-z0-9]$");
+
+#[cfg(feature = "wasm")]
+pub mod wasm {
+    pub fn is_dns_address(s: &str) -> bool {
+        crate::network::is_dns_address_regex(s)
```

Comment:
Regex is a better approach than socket check IMO. The domain does not have to be immediately resolvable, or does it?

Reply:
Regex will always be only partially correct. I’d rather have the info whether my configuration is broken early than watching logs.